plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/product-edit.ts

```diff
@@ -6,10 +6,16 @@ import { recordEvent } from '@woocommerce/tracks';
 /**
  * Internal dependencies
  */
-import { addExitPageListener, initProductScreenTracks } from './shared';
+import {
+	addExitPageListener,
+	getProductData,
+	initProductScreenTracks,
+} from './shared';
 
 const initTracks = () => {
-	recordEvent( 'product_edit_view' );
+	const productData = getProductData();
+
+	recordEvent( 'product_edit_view', { product_id: productData?.product_id } );
```

Comment:
The current solution is good. I lean towards deconstructing the const, but it's totally ok as it is.

```
const { product_id } = getProductData();
recordEvent( 'product_edit_view', { product_id } );
```